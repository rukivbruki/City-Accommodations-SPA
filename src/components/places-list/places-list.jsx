import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import PlaceCard from "../place-card/place-card";
import {ActionCreator} from "../../reducer/data/data";

class PlacesList extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {offers, changeActiveOffer} = this.props;

    return (
      <div className="cities__places-list places__list tabs__content">
        {offers.map((offer) =>
          <PlaceCard
            key={offer.id}
            offer={offer}
            onCardHover={changeActiveOffer}
          />)}
      </div>
    );
  }
}

PlacesList.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    city: PropTypes.shape({
      name: PropTypes.string.isRequired,
      location: PropTypes.arrayOf(PropTypes.number).isRequired,
      zoom: PropTypes.number.isRequired
    }),
    location: PropTypes.shape({
      coordinates: PropTypes.arrayOf(PropTypes.number).isRequired,
      zoom: PropTypes.number.isRequired
    }),
    title: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    previewImage: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    isPremium: PropTypes.bool.isRequired,
    isFavorite: PropTypes.bool.isRequired,
  })).isRequired,
  changeActiveOffer: PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch) => ({
  changeActiveOffer: (id) => dispatch(ActionCreator.changeActiveOffer(id))
});

export {PlacesList};
export default connect(null, mapDispatchToProps)(PlacesList);
