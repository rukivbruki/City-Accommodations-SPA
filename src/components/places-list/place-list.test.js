import React from "react";
import renderer from "react-test-renderer";
import {PlacesList} from "./places-list";

jest.mock(`../place-card/place-card.jsx`, () => jest.fn().mockReturnValue(null));

it(`PlacesList component render correct`, () => {
  const OFFERS = [
    {
      id: 1,
      city: {
        name: `Hamburg`,
        location: [53.570341000000006, 9.975654],
        zoom: 12
      },
      title: `Amazing and Extremely Central Flatn`,
      previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      isPremium: false,
      isFavorite: false,
      price: 103,
      rating: 4.3,
      type: `room`,
    },
    {
      id: 2,
      city: {
        name: `Paris`,
        location: [48.846610000000005, 2.374499],
        zoom: 12
      },
      title: `Canal View Prinsengracht`,
      previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      price: 109,
      isPremium: false,
      isFavorite: false,
      rating: 4,
      type: `room`,
    },
  ];

  const app = renderer
    .create(
        <PlacesList
          offers={OFFERS}
          changeActiveOffer={jest.fn()}
          onSelect={jest.fn()}
        />)
    .toJSON();

  expect(app).toMatchSnapshot();
});
