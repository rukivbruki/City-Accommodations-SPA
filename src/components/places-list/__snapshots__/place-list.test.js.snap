// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesList component render correct 1`] = `
<div
  className="cities__places-list places__list tabs__content"
/>
`;
